packages/react-kit/src/utils/with-rx2.ts

```diff
@@ -44,11 +42,13 @@ export const withRX = <P extends object>(Target: ComponentType<P>) => <D extends
 					props[key].pipe(map((value: unknown) => ({ [key]: value }))),
 				);
 				this.inputSubscription = merge(...inputs)
-					.pipe(observeOn(scheduler))
+					.pipe(options.scheduler ? observeOn(options.scheduler) : identity)
 					.subscribe(this.setState.bind(this));
 			}
 			if (effects$) {
-				this.effectsSubscription = effects$.pipe(observeOn(scheduler)).subscribe();
+				this.effectsSubscription = effects$
+					.pipe(options.scheduler ? observeOn(options.scheduler) : identity)
```

Comment:
It looks like we don't need `observeOn` here at all

Reply:
Why so? Do tests run green without it?

Reply:
I forgot about the tests. If unit tests require custom scheduler, then this should stay for sure. But then, as above, I don't quite like `pipe(identity)` thing.